The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11509892266/job/32040800657) [(plain)](https://github.com/rust-lang/rust/commit/89bbecd9e96f29577a742eee7cc937a1bcf172bc/checks/32040800657/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.883 Building wheels for collected packages: reuse
#13 2.884   Building wheel for reuse (pyproject.toml): started
#13 3.127   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.128   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.128   Stored in directory: /tmp/pip-ephem-wheel-cache-k2wcdlwe/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.131 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.518 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.518 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.036 Collecting virtualenv
#13 4.036 Collecting virtualenv
#13 4.106   Downloading virtualenv-20.27.0-py3-none-any.whl (3.1 MB)
#13 4.202      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 33.5 MB/s eta 0:00:00
#13 4.259 Collecting filelock<4,>=3.12.2
#13 4.273   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.291 Collecting distlib<1,>=0.3.7
#13 4.306   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.313      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.9 MB/s eta 0:00:00
#13 4.347 Collecting platformdirs<5,>=3.9.1
#13 4.361   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.441 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.613 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.0
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      178112 kB
DirectMap2M:     7161856 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARNING: `rust.download-rustc` is enabled. The `rust.channel` option will be overridden by the CI rustc's channel.
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt check
Diff in /checkout/src/librustdoc/html/render/search_index.rs:759:
         }
     });
     let (mut inputs, mut output, where_clause) = match item.kind {
-        clean::ForeignFunctionItem(ref f, _) |
-        clean::FunctionItem(ref f) | clean::MethodItem(ref f, _) | clean::TyMethodItem(ref f) => {
+        clean::ForeignFunctionItem(ref f, _)
+        | clean::FunctionItem(ref f)
+        | clean::MethodItem(ref f, _)
+        | clean::TyMethodItem(ref f) => {
             get_fn_inputs_and_outputs(f, tcx, impl_or_trait_generics, cache)
         _ => return None,
         _ => return None,
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/librustdoc/html/escape.rs" "/checkout/src/librustdoc/html/length_limit/tests.rs" "/checkout/src/librustdoc/html/url_parts_builder/tests.rs" "/checkout/src/librustdoc/html/format.rs" "/checkout/src/librustdoc/html/tests.rs" "/checkout/src/librustdoc/html/markdown/tests.rs" "/checkout/src/librustdoc/html/markdown/footnotes.rs" "/checkout/src/librustdoc/html/escape/tests.rs" "/checkout/src/librustdoc/html/highlight/tests.rs" "/checkout/src/librustdoc/html/highlight/fixtures/sample.rs" "/checkout/src/librustdoc/html/highlight/fixtures/union.rs" "/checkout/src/librustdoc/html/url_parts_builder.rs" "/checkout/src/librustdoc/html/layout.rs" "/checkout/src/librustdoc/html/toc/tests.rs" "/checkout/src/librustdoc/html/static_files.rs" "/checkout/src/librustdoc/html/highlight.rs" "/checkout/src/librustdoc/html/mod.rs" "/checkout/src/librustdoc/html/toc.rs" "/checkout/src/librustdoc/html/render/ordered_json.rs" "/checkout/src/librustdoc/html/render/sorted_template/tests.rs" "/checkout/src/librustdoc/html/render/span_map.rs" "/checkout/src/librustdoc/html/render/context.rs" "/checkout/src/librustdoc/html/render/ordered_json/tests.rs" "/checkout/src/librustdoc/html/render/sorted_template.rs" "/checkout/src/librustdoc/html/render/print_item.rs" "/checkout/src/librustdoc/html/render/sidebar.rs" "/checkout/src/librustdoc/html/render/type_layout.rs" "/checkout/src/librustdoc/html/render/write_shared.rs" "/checkout/src/librustdoc/html/render/tests.rs" "/checkout/src/librustdoc/html/render/search_index/encode.rs" "/checkout/src/librustdoc/html/render/search_index.rs" "/checkout/src/librustdoc/html/render/mod.rs" "/checkout/src/librustdoc/html/render/write_shared/tests.rs" "/checkout/src/librustdoc/html/sources.rs" "/checkout/src/librustdoc/lint.rs" "/checkout/src/librustdoc/visit_lib.rs" "/checkout/src/librustdoc/lib.rs" "/checkout/src/librustdoc/doctest/markdown.rs" "/checkout/src/librustdoc/doctest/runner.rs" "/checkout/src/librustdoc/doctest/rust.rs" "/checkout/src/librustdoc/doctest/make.rs" "/checkout/src/librustdoc/doctest/tests.rs" "/checkout/src/librustdoc/formats/cache.rs" "/checkout/src/librustdoc/formats/item_type.rs" "/checkout/src/librustdoc/formats/renderer.rs" "/checkout/src/librustdoc/formats/mod.rs" "/checkout/src/librustdoc/theme/tests.rs" "/checkout/src/librustdoc/error.rs" "/checkout/src/librustdoc/visit_ast.rs" "/checkout/src/librustdoc/html/length_limit.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Oct 25 01:02:20 UTC 2024
  network time: Fri, 25 Oct 2024 01:02:20 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/html/render/search_index.rs
+++ b/src/librustdoc/html/render/search_index.rs
@@ -759,7 +759,10 @@ pub(crate) fn get_function_type_for_search<'tcx>(
         }
     });
     let (mut inputs, mut output, where_clause) = match item.kind {
-        clean::FunctionItem(ref f) | clean::MethodItem(ref f, _) | clean::TyMethodItem(ref f) => {
+        clean::ForeignFunctionItem(ref f, _)
+        | clean::FunctionItem(ref f)
+        | clean::MethodItem(ref f, _)
+        | clean::TyMethodItem(ref f) => {
             get_fn_inputs_and_outputs(f, tcx, impl_or_trait_generics, cache)
         }
         _ => return None,

--- a/tests/rustdoc-js/extern-func.js
+++ b/tests/rustdoc-js/extern-func.js
@@ -1,0 +1,8 @@
+const EXPECTED = [
+    {
+        'query': 'c_float -> c_float',
+        'others': [
+            { 'path': 'extern_func', 'name': 'sqrt' }
+        ],
+    },
+];

--- a/tests/rustdoc-js/extern-func.rs
+++ b/tests/rustdoc-js/extern-func.rs
@@ -1,0 +1,5 @@
+use std::ffi::c_float;
+
+extern "C" {
+    pub fn sqrt(x: c_float) -> c_float;
+}

